src/Framework/TestCase.php

```diff
@@ -1741,6 +1751,48 @@ private function compareGlobalStateSnapshotPart(array $before, array $after, str
         }
     }
 
+    private function handleEnvironmentVariables(): void
+    {
+        $withEnvironmentVariables = MetadataRegistry::parser()->forClassAndMethod(static::class, $this->methodName)->isWithEnvironmentVariable();
+
+        $environmentVariables = [];
+
+        foreach ($withEnvironmentVariables as $metadata) {
+            assert($metadata instanceof WithEnvironmentVariable);
+
+            $environmentVariables[$metadata->environmentVariableName()] = $metadata->value();
+        }
+
+        foreach ($environmentVariables as $environmentVariableName => $environmentVariableValue) {
+            assert($metadata instanceof WithEnvironmentVariable);
+
+            $this->backupEnvironmentVariables[$environmentVariableName] = $_ENV[$environmentVariableName] ?? getenv($environmentVariableName);
```

Comment:
I was torn on this... in PHP, nothing prevents to have a different value for `$_ENV['VAR']` and `getenv('VAR')`.
Do you think I should store both representations, in order to have 100% the same environment afterand before the test?

Reply:
I think so, yes.

Reply:
any hints how I should store this?
my first guess would be to store this in an array:
```php
    /**
     * @var array{getenv: array<string, false|string>, global: array<string, false|string>}
     */
    private array $backupEnvironmentVariables = ['getenv' => [], 'global'  => []];
```

but I don't like it. Another idea would be to use a value object:
```php
final readonly class BackupEnvVar
{
    public static function asGetenv(): self;
    public static function asSuperGlobal(): self;    
}
```

Reply:
I've updated the code to use a value object. (some tests are failing, I'll check them after we agree on the implementation)